### PR TITLE
finished CBoat

### PR DESCRIPTION
**Pull request by aap**

### CODE REVIEW

File: src/vehicles/Boat.cpp [hunk 1]
```diff
@@ -94,6 +99,541 @@ CBoat::GetComponentWorldPosition(int32 component, CVector &pos)
...
+			CVector splashImpulse = speed * m_fMass;
+			ApplyMoveForce(splashImpulse);
+			ApplyTurnForce(splashImpulse, buoyancePoint);
+		}
+
+		// Spray particles on sides of boat
+		if(m_nDeltaVolumeUnderWater > 75){
+			float speed = m_vecMoveSpeed.Magnitude();
+			float splash1Size = speed;
+			float splash2Size = m_nDeltaVolumeUnderWater * 0.005 * 0.2f;
```
Comment (ShFil119) [inline src/vehicles/Boat.cpp:422]: nit 0.005f

File: src/vehicles/Boat.cpp [hunk 2]
```diff
@@ -275,6 +814,43 @@ CBoat::FillBoatList()
 	}
 }
 
+void
+CBoat::PruneWakeTrail(void)
+{
+	int i;
+
+	for(i = 0; i < 32; i++){
```
Comment (ShFil119) [inline src/vehicles/Boat.cpp:822]: `ARRAY_SIZE(m_afWakePointLifeTime)`

File: src/vehicles/Boat.cpp [hunk 3]
```diff
@@ -275,6 +814,43 @@ CBoat::FillBoatList()
...
+	m_nNumWakePoints = i;
+}
+
+void
+CBoat::AddWakePoint(CVector point)
+{
+	int i;
+	if(m_afWakePointLifeTime[0] > 0.0f){
+		if((CVector2D(GetPosition()) - m_avec2dWakePoints[0]).MagnitudeSqr() < SQR(1.0f)){
+			for(i = min(m_nNumWakePoints, 32-1); i != 0; i--){
```
Comment (ShFil119) [inline src/vehicles/Boat.cpp:840]: also

**Comment by erorcun** [comment]
lgtm

**Pull request merged**